### ISSUE TITLE
Error's are reported at wrong column number when tabs are used.

### ISSUE THREAD

**Issue by coreyfarrell** [eslint/eslint#8386]
**Tell us about your environment**

* **ESLint Version: v3.19.0**
* **Node Version: v6.10.0**
* **npm Version: v3.10.10**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
{
    "extends": "eslint:recommended",
    "rules": {
        "indent": ["error", "tab"],
        "max-len": ["error", 120, 4],
        "no-useless-escape": ["error"],
        "no-unused-vars": "off"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
(function() {
	'use strict';
	var regex = (/\.[\.0]/);
})();
```

**What did you expect to happen?**
Errors should indicate source locations matching what would be shown in an editor, not "byte" position.  In my editor tab-size is 4 columns.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/user/test/test.js
  3:19  error  Unnecessary escape character: \.            no-useless-escape

✖ 1 problems (1 errors, 0 warnings)
```

According to any editor with tab-size=4, this error is at line 3 column 22.  It appears that tab-size is not picked up from the `max-len` rule.  I used the `no-useless-escape` rule to show how this could be significant as regular expressions are often ugly the exact column can be important.

**Comment by not-an-aardvark** [comment]
Thanks for the report.

However, this is working as intended. ESLint's columns are intentionally based on the number of characters from the start of the column, because in general ESLint doesn't know what tab width the user would want.

(You suggested picking up the tab width from the `max-len` rule, but the user won't always have that rule enabled. So it wouldn't make sense to modify error locations for all other rules based on the `max-len` rule's configuration, because that configuration might not exist.)

If your editor integration isn't displaying errors in the correct location, I would recommend filing a bug with the editor integration.

**Comment by coreyfarrell** [comment]
I understand what you mean by editor integration and I've since installed VSCode integration which does work nicely / correctly.  The work-flow I was reporting is that I run `eslint` from a terminal window, then manually scrolling to the reported position in any generic editor to make the code fix.

Honestly I'm still getting used to the concept of an integrated IDE, most of my work is still done bouncing back and forth between the editor and a terminal window.  If this is a "won't fix" type of issue I understand, but when running from command-line it seems like tabs should not be reported as 1 column wide.  I realize `max-len` is the wrong place to find this information, that was just a stab in the dark / the only way I could find to tell `.eslintrc.json` how big a tab is.  I assume for the command-line output to be fixed would require a new generic setting for tab-width.

**Comment by not-an-aardvark** [comment]
Have you tried using a different [formatter](http://eslint.org/docs/user-guide/command-line-interface#-f---format)? For example, if you use the `codeframe` formatter (by using `--format codeframe` on the command line), you can get an error report that points to the location in the file where the problem occurs. (Although now that I think about it, I'm not sure whether the `codeframe` handles tabs well either -- if that's the case, we should probably fix it.) You can also create a custom formatter to display problems according to your preferred tab width.

**Comment by coreyfarrell** [comment]
`codeframe` does visually handle tabs correctly, the `^` is pointing at the character which starts the error.  Looks like the prefix to `^` uses tabs where appropriate, so the terminal is visually correct.  Of course it still states the incorrect numeric position as it doesn't know how to count tabs:
```
error: Unnecessary escape character: \. (no-useless-escape) at test.js:3:19:
  1 | (function() {
  2 | 	'use strict';
> 3 | 	var regex = (/\.[\.0]/);
    | 	                 ^
  4 | })();
  5 | 


1 error found.
```

**Comment by not-an-aardvark** [comment]
Hmm, ok. This is working as intended, but you can always create a custom formatter if you want to display a different column number.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
